src/graphics/canvas.rs

```diff
@@ -1,388 +1,526 @@
-        self.image.blend_mode = mode;
+
+    /// Draws a rectangle instanced multiple times, as defined by the given [`InstanceArray`].
+    ///
+    /// Also see [`Canvas::draw_mesh_instances()`].
+    pub fn draw_instances(&mut self, instances: InstanceArray, param: impl Into<DrawParam>) {
+        let param = param.into();
+        self.draws.entry(param.z).or_default().push(DrawCommand {
+            state: self.state.clone(),
+            draw: Draw::MeshInstances {
+                mesh: self.defaults.mesh.clone(),
+                instances,
+                param,
+            },
+        });
     }
-    fn blend_mode(&self) -> Option<BlendMode> {
-        self.image.blend_mode
+
+    /// Draws a section text that is fit and aligned into a given `rect` bounds.
+    ///
+    /// The section can be made up of multiple [Text], letting the user have complex formatting
+    /// in the same section of text (e.g. bolding, highlighting, headers, etc).
+    ///
+    /// [TextLayout] determines how the text is aligned in `rect` and whether the text wraps or not.
+    ///
+    /// ## A tip for performance
+    /// Text rendering will automatically batch *as long as the text draws are consecutive*.
+    /// As such, to achieve the best performance, do all your text rendering in a single burst.
+    pub fn draw_bounded_text(
+        &mut self,
+        text: &[Text],
+        rect: Rect,
+        rotation: f32,
+        layout: TextLayout,
+        z: ZIndex,
+    ) {
+        self.draws.entry(z).or_default().push(DrawCommand {
+            state: self.state.clone(),
+            draw: Draw::BoundedText {
+                text: text.to_vec(),
+                rect,
+                rotation,
+                layout,
+            },
+        });
     }
-}
 
-fn flip_draw_param_vertical(param: DrawParam) -> DrawParam {
-    let param = if let Transform::Matrix(mat) = param.trans {
-        param.transform(
-            glam::Mat4::from(mat)
-                * glam::Mat4::from_scale_rotation_translation(
-                    glam::vec3(1.0, -1.0, 1.0),
-                    Quat::IDENTITY,
-                    glam::vec3(0.0, 1.0, 0.0),
-                ),
+    /// Unbounded version of [`Canvas::draw_bounded_text()`].
+    pub fn draw_text(
+        &mut self,
+        text: &[Text],
+        pos: impl Into<mint::Vector2<f32>>,
+        rotation: f32,
+        layout: TextLayout,
+        z: ZIndex,
+    ) {
+        let pos = pos.into();
+        self.draw_bounded_text(
+            text,
+            Rect::new(pos.x, pos.y, f32::INFINITY, f32::INFINITY),
+            rotation,
+            layout,
+            z,
         )
-    } else {
-        panic!("Can not be called with a non-matrix DrawParam");
-    };
-    let new_src = Rect {
-        x: param.src.x,
-        y: (1.0 - param.src.h) - param.src.y,
-        w: param.src.w,
-        h: param.src.h,
-    };
-    param.src(new_src)
+    }
```

Comment:
I'm wondering whether adding a `DrawParam`-like struct (`TextDrawParam` or something) might be worth it for having similar benefits (having comfortable ergonomics where you can emid most things, as they have default values).
But there are only 4 parameters and 1 of them will pretty much always be set over the default (pos), so it would only shorten it by the other 3 parameters. Really not sure whether that's worth the weight of creating/maintaining `TextDrawParam` though.

Reply:
`TextDrawParam` would be useful. Shouldn't it be possible to impl From<DrawParam> for TextDrawParam if that's added.

Reply:
That would be possible, but From/Into doesn't work transitively, meaning that we wouldn't get access to all the nice `From` implementations for `DrawParam`. A user would have to explicitly create a `DrawParam`, just to `into` it to a `TextDrawParam`. So if we want the same level of comfort for `TextDrawParam` we'd still have to implement `From` nicely there as well.